File: src/main/java/ku/cse/team11/RankHub/domain/Content.java
package ku.cse.team11.RankHub.domain;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.LocalDateTime;

@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class Content {
    @Id
    // @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // 기본 정보
    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String authors;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ContentType contentType;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Platform platform;

    private String thumbnailUrl;

    @Column(nullable = false)
    private String contentUrl;

    private Integer totalEpisodes;

    // 초기만 문자열
    @Column(columnDefinition = "TEXT")
    private String tags;

    private String category;

    private String ageRating;

    private String pubPeriod;

    // 통계
    private Long views = 0L;

    private Double rating = 0.0;

    private Long likes = 0L;

    @CreationTimestamp
    private LocalDateTime createdAt;

    @UpdateTimestamp
    private LocalDateTime updatedAt;
}

File: app/src/main/java/com/example/ku_cse_team11_mobileapp/api/model/Content.kt
package com.example.ku_cse_team11_mobileapp.api.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

// 랭킹 API 응답
data class RankResponse(
    val rank: Int,
    val content: ContentSummary
)

// 콘텐츠 요약 정보 (랭킹용)
data class ContentSummary(
    val id: Int,
    val title: String,
    val authors: String,
    val thumbnailUrl: String?,
    val platform: String,
    val views: Long,
    val language: String? = null,
    val tier: String? = null

)

// 콘텐츠 상세 응답
data class ContentDetail(
    val id: Int,
    val title: String,
    val authors: String,
    val contentType: String,
    val description: String,
    val platform: String,
    val thumbnailUrl: String?,
    val contentUrl: String?,
    val totalEpisodes: Int,
    val tags: String,
    val category: String,
    val ageRating: String,
    val pubPeriod: String,
    val views: Long,
    val rating: Double,
    val likes: Int,
    val createdAt: String,
    val updatedAt: String,
    val language: String? = null,
    val myTier: String? = null,            // ✅ 추가
    val avgTier: String? = null,           // ✅ 추가
    val stats: TierStats? = null
)

@Serializable
data class TierStats(
    val rating: Map<String, Int> = emptyMap(),   // e.g. {"A":1,"B":1,"S":0,"C":0,"D":0}
    @SerialName("rating_count") val ratingCount: Int = 0
)